src/components/wallet/views/claimableDisplay.tsx

```diff
@@ -0,0 +1,58 @@
+import { h } from '@stencil/core'
+import { has as loHas } from 'lodash-es'
+import WalletButton from './walletButton'
+import { Wallet } from '../wallet'
+
+interface ClaimableBalance {
+  id: string
+  asset: string
+  amount: string
+  sponsor?: string
+  claimants: any[]
+}
+
+export default function claimableDisplay(this: Wallet) {
+  if (this.account.availableBalances) {
+    const claimableRecords = (claimableBalances: any) => {
+      return claimableBalances.records.map(claimableBalanceRow)
+    }
+    const claimableBalanceRow = (balance: ClaimableBalance) => {
+      const loadingKey = (type: string) => {
+        return `${type}:${balance.asset.split(':')[0]}:${balance.id}`
+      }
+      const assetCode = balance.asset.split(':')[0]
+      const claimBalanceButton = WalletButton.call(
+        this,
+        'Claim',
+        loadingKey('claim'),
+        () => {
+          this.claimAsset(
+            balance.id,
+            assetCode,
+            balance.sponsor,
+            balance.amount
+          )
+        }
+      )
+      return (
+        <div class="asset-row">
+          <div class="balance-row">
+            <div class="asset-code">{assetCode}:</div>
+            <div class="balance">{balance.amount}</div>
+          </div>
+          <div class="actions">{claimBalanceButton}</div>
+        </div>
+      )
+    }
+
+    return (
+      <div>
+        {loHas(this.account, 'claimableBalances') ? (
```

Comment:
What is `loHas`? I've never understood the `lo*` functions

Reply:
And when is `this.account.claimableBalances` populated?

Reply:
the lo* functions are used to check, manipulate, etc (ie the [lodash](https://lodash.com/) functions) are used to make it "easier" to interface with json items primarily in this codebase.
It is nice since it will look for really nested keys.
so `loHas(this.account, 'claimableBalances')` is a way to check the existence of a claimableBalances key

Reply:
`claimableBalances` is populated updated etc in during the updateAccount process.
Which happens frequently and is populated before claimable balances are displayed.